node/resolvers/profile/index.ts

```diff
@@ -44,6 +52,37 @@ export const mutations = {
 export const queries = {
   profile: (_: any, { customFields }: any, context: Context) =>
     getProfile(context, customFields),
+
+  checkProfileAllowed: async (_: any, __: any, context: Context) => {
+    const {
+      clients: { catalog, session },
+      vtex: { segment },
+      cookies,
+    } = context
+    const salesChannel = segment ? segment.channel : null
+
+    const { sessionData } = await session.getSession(
+      cookies.get('vtex_session')!,
+      ['*']
+    )
+
+    const email: string | undefined = path(
+      ['namespaces', 'profile', 'email', 'value'],
+      sessionData
+    )
+
+    if (!email) {
+      return { allowed: null }
+    }
+
+    const availableSalesChannels = await catalog
+      .salesChannelAvailable(email)
+      .catch(() => [])
```

Comment:
```suggestion
      .catch(() => {})
```

Reply:
Nope, that doesnt have the same effect.

The API returns an array of available sales channels. If the API fails, I want to respond that the user doesnt have access to any sales channel.